apps/prairielearn/src/components/DeleteCourseInstanceModal.tsx

```diff
@@ -0,0 +1,194 @@
+                  <input
+                    class="form-check-input"
+                    type="checkbox"
+                    id="irreversible"
+                    checked={checks.irreversible}
+                    onChange={(e) =>
+                      setChecks((c) => ({ ...c, irreversible: e.currentTarget.checked }))
+                    }
+                  />
+                  <label class="form-check-label" for="irreversible">
+                    I understand that once deleted, this course instance cannot be restored through
+                    the PrairieLearn web interface.
+                  </label>
+                </div>
+                <div class="form-check mb-2">
+                  <input
+                    class="form-check-input"
+                    type="checkbox"
+                    id="recovery"
+                    checked={checks.recovery}
+                    onChange={(e) =>
+                      setChecks((c) => ({ ...c, recovery: e.currentTarget.checked }))
+                    }
+                  />
+                  <label class="form-check-label" for="recovery">
+                    I understand data recovery requires manual Git operations on the course
+                    repository.
+                  </label>
+                </div>
+              </div>
+            )}
+            {step === 2 && (
+              <div>
+                <div class="card text-bg-light mb-3">
+                  <div class="card-body">
+                    <h3 class="h5 card-title">Recovery information</h3>
+                    <p>If you need to recover this course instance, you will need to:</p>
+                    <ol class="mb-0">
+                      <li>Access the course's Git repository</li>
+                      <li>
+                        Use Git commands to restore the course instance from the commit history
+                      </li>
+                      <li>Push your changes</li>
+                      <li>Sync your changes to PrairieLearn</li>
+                    </ol>
+                  </div>
+                </div>
+                <div>
+                  <label htmlFor="confirmShortName" class="form-label">
+                    Type <strong>{shortName}</strong> to confirm deletion:
+                  </label>
+                  <input
+                    ref={confirmationTextRef}
+                    id="confirmShortName"
+                    name="confirmShortName"
+                    class="form-control"
+                    value={confirmationText}
+                    autoComplete="off"
+                    onInput={(e) => setConfirmationText(e.currentTarget.value)}
+                  />
+                </div>
+              </div>
+            )}
+          </div>
+          <div class="modal-footer">
+            {step === 1 && (
+              <>
+                <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">
```

Comment:
nit: you can pull this button out of the steps since it's the same and avoid a fragment for the second element as a benefit.

Reply:
Nice idea, done in 41f7ea9.